fe/fe-core/src/main/java/org/apache/doris/analysis/ArithmeticExpr.java

```diff
@@ -536,7 +536,7 @@ private void analyzeDecimalV3Op(Type t1, Type t2) throws AnalysisException {
     public void analyzeImpl(Analyzer analyzer) throws AnalysisException {
         if (VectorizedUtil.isVectorized()) {
             for (Expr child : children) {
-                if (child instanceof DecimalLiteral && child.getType().isDecimalV3()) {
+                if (child instanceof DecimalLiteral && child.getType().isDecimalV3() && !child.getIsCasted()) {
                     ((DecimalLiteral) child).tryToReduceType();
```

Comment:
try sql `select 1.0000000 / cast(3 as DECIMALV3(1, 0));` seems this pr can't handle such case, IMO,`tryToReduceType()` method is ambiguous, it's not clear why need this function, is it possible to remove it?

Reply:
I asked the author about this code, and they said it was meant to reduce precision during calculations.  now it seems better to remove it.